[Siddardha Reddy Sanagala] iP

Siddardar opened the pull request:
## Tars

>[TARS] [as Cooper repairs him] Settings. General settings. Security settings.
>[TARS]: Honesty, new setting: ninety-five percent.
>[TARS]: Confirmed. Additional settings.
>[Cooper]: Humor, seventy-five percent.
>[TARS]: Confirmed. Self destruct sequence in T minus 10, 9...
>[Cooper]: Let's make that sixty percent.
>[TARS]: Sixty percent, confirmed. Knock knock.
>[Cooper]: You want fifty-five?

Tars is a chat bot that manages tasks and with a slightly sarcastic and humorous twist much like it's namesake in [Interstellar](https://www.imdb.com/title/tt0816692/). (Great movie btw you should definitely check it out if you haven't 🙂)

Features:
- Text-based
- Easy to learn
- ~~FAST~~ _**SUPER** FAST_ to use!

And the best part is that it is **FREE!!!**

All you need to do is:
1. Download tars.jar from [here](https://github.com/Siddardar/ip/releases/tag/A-Jar)
2. Double-click it. 
3. Add your tasks
4. Let it manage your tasks for you

It can:
- [X] Manage todos, deadlines and events
- [ ] interact via a GUI (coming soon!)

If you are familiar with Java, feel free to modify the source code. Here is the `main` method:
```java
public static void main(String[] args) {

    Tars tars = new Tars();
    tars.run();
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

Siddardar committed:
Level 0

Siddardar committed:
Level 1

Siddardar committed:
Implemented add to list functionality

Siddardar committed:
Implemented add to list functionality

Siddardar committed:
Minor bug fixes

Siddardar committed:
Implemented mark and unmark functionality

Siddardar committed:
Implemented different types of tasks: ToDo, Deadline and Event

Siddardar committed:
Implemented automated testing

Siddardar committed:
Implemented exception handling

Siddardar committed:
Implemented delete functionality

Siddardar committed:
Add read/write functionality

Siddardar committed:
Add DateTime formats to Deadline.java and Event.java

Siddardar committed:
Merge branch 'branch-Level-8' into master

Siddardar committed:
Add classes to support OOP

Siddardar committed:
Merge branch 'branch-A-More-OOP' into master

Siddardar committed:
Change .gitignore to exclude file tasks are saved to on local machine

Siddardar committed:
Move java files into packages

Siddardar committed:
Merge branch 'branch-A-Packages' into master

Siddardar committed:
Merge branch 'branch-A-Gradle' with master

Siddardar committed:
Add gradle wrapper to chatbot

Siddardar committed:
Add JUnit testing for commands

Siddardar committed:
Merge branch 'branch-A-JUnit' with master

Siddardar committed:
Remove cache to stop tracking of tarsTasks.txt

Siddardar committed:
Edit files to comply with coding standards

Siddardar committed:
Add JavaDocs to all current methods and classes

Siddardar committed:
Add find functionality to chatbot

Siddardar committed:
Merge branch 'branch-A-CodingStandard' with master

Siddardar committed:
Resolve merge conflict when merging branch-A-JavaDoc with master

Siddardar committed:
Fix more merge conflicts when merging branch-A-JavaDoc with master

Siddardar committed:
Resolve merge conflicts when merging branch-Level-9 with master

Siddardar committed:
Add GUI for Tars

Siddardar committed:
Edit build.gradle to package app as JAR file and fix minor UI bugs in Response.java

charlesliyifeng commented on src/main/java/tars/Response.java:
```diff
@@ -0,0 +1,120 @@
+    /**
+     * Displays the introductory message when the TARS application starts.
+     */
+    public void intro() {
```
Maybe use verbs for method names?

charlesliyifeng commented on src/main/java/tars/Response.java:
```diff
@@ -0,0 +1,120 @@
+ * various command classes ({@link MarkCommand}, {@link DeleteCommand}, {@link AddCommand})
+ * to execute user commands and generate appropriate responses.</p>
+ */
+public class Response {
```
Maybe it would be more managable to separate the UI and the parser into separate classes?

charlesliyifeng commented on src/main/java/tars/commands/FindCommand.java:
```diff
@@ -0,0 +1,40 @@
+
+        List<String> tasksFound = tasks.findTaskByDescription(input);
+
+        StringBuilder str = new StringBuilder("Here are the tasks found.\n");
```
Maybe use a more descriptive variable name?

charlesliyifeng commented on src/main/java/tars/parsers/DeadlineParser.java:
```diff
@@ -0,0 +1,90 @@
+            throw new TarsException("Add a name to your deadline");
+        }
+
+        String[] split = taskInfo[1].split("/", 2);
```
Perhaps you should not use the same name as a method for split?

charlesliyifeng commented on src/main/java/tars/parsers/DeadlineParser.java:
```diff
@@ -0,0 +1,90 @@
+        }
+
+        switch(deadlineCommand.length) {
+            case 1:
```
This switch statement is not following the coding standard.

charlesliyifeng reviewed (commented):
Interesting code structure. LGTM except for some minor issues with variable names to fix.

Siddardar committed:
Merge branch 'branch-Level-10' with master

youngseopark05 commented on build.gradle:
```diff
@@ -0,0 +1,41 @@
+    archiveClassifier = null
+}
+
+run{
```
Should there be a whitespace before the braces here?

youngseopark05 commented on src/main/java/tars/Response.java:
```diff
@@ -0,0 +1,120 @@
+                formatResponse(e.getMessage());
+
+            }
+        }
```
Would it be more suitable to turn this into a switch-case statement?

youngseopark05 commented on src/main/java/tars/Response.java:
```diff
@@ -0,0 +1,120 @@
+            } catch (TarsException e) {
+                formatResponse(e.getMessage());
+            }
+
```
Is this whitespace necessary?

youngseopark05 commented on src/main/java/tars/parsers/Parser.java:
```diff
@@ -0,0 +1,30 @@
+ * and requires subclasses to implement the {@link #parse(String[])} method
+ * to handle specific parsing logic.</p>
+ */
+public abstract class Parser {
```
I like this idea of having different Parsers for the different types of tasks, I might implement it into my own iP as well.

youngseopark05 commented on src/main/java/tars/parsers/EventParser.java:
```diff
@@ -0,0 +1,135 @@
+        if (startCommand == null) {
+            throw new TarsException("Add a /from command and a start date");
+        } else {
+            switch (startCommand.length) {
+                case 1:
+                    if (startCommand[0].equals("from")) {
+                        throw new TarsException("Add a event start date");
+                    } else {
+                        throw new TarsException("Add the /from command");
+                    }
+
+                case 2:
+                    if (startCommand[0].equals("from")) {
+                        if (startCommand[1].isEmpty()) {
+                            throw new TarsException("Add an event start date");
+                        }
+                    } else {
+                        throw new TarsException("Add the /from command");
+                    }
+            }
```
Should the case statements be on the same indent level as the switch clause?

youngseopark05 reviewed (commented):
Great job overall, I can't find much coding style violations at this stage, except for a wrongly indented switch-case statements and unnecessary whitespaces. Tracking and organising the whitespaces and empty lines might take some time, but I think once the whitespaces are cleaned up it will add to the readability!

Siddardar committed:
Add assertions to code

Siddardar committed:
Improve code quality

Siddardar committed:
Refactor findTasks method to use Java Streams

The findTasks method currently uses a for loop to iterate
through the task list and filter tasks. Using Streams
simplifies the code, making it more readable and concise.

Extracting the behavior of the for loop into a Stream
operation allows us to perform the same logic in a more
declarative style, improving code clarity and maintainability.

Switching to Streams is preferable as it leverages Java's
functional programming features, reducing the potential
for errors and making future enhancements easier.

Siddardar committed:
Merge pull request #1 from branch-A-Assertions

Add assertions to code

Siddardar committed:
Merge branch 'master' into branch-A-CodeQuality

Siddardar committed:
Merge pull request #2 from branch-A-CodeQuality

Improve code quality

Siddardar committed:
Merge pull request #3 from branch-A-Streams

Refactor findTasks method to use Java Streams

Siddardar committed:
Add workflow file

Siddardar committed:
Merge branch branch-A-CI with master

Siddardar committed:
Add workflow directory

Siddardar committed:
Add sorting to tasks in TaskList

Siddardar committed:
Merge branch branch-C-Sort with master

Siddardar committed:
Add User Guide documentation and update GUI design

User Guide lacks detailed instructions for commands, and the
GUI has resizing issues that affect usability.

Clear documentation is needed to help users understand and use
commands, and the GUI needs improvements to be more
user-friendly.

Add new sections to the User Guide for command instructions and
update the GUI layout by making it resizable and add colours
throughout the GUI to make it more visually appealing.

Updating the User Guide and refining the GUI design together keeps
documentation consistent with the interface, reducing confusion.

Siddardar committed:
Fix user guide formatting

Siddardar committed:
Remove redundancies in FXML code

Siddardar committed:
Fix bugs in test cases

Siddardar committed:
Update README.md

Siddardar committed:
Improve command validation

Siddardar committed:
Fix bug with user guide summary table

Siddardar committed:
Remove command summary header

Siddardar committed:
Fix formatting issues in user guide

Siddardar committed:
Improve error handing for delete command

Siddardar committed:
Improve method names of tests

Siddardar committed:
Add more unit testing
